### ISSUE TITLE
Pretty-printing

### ISSUE THREAD

**Issue by effectfully** [effectfully/tiny-lang#30]
Currently we just dump expressions as `String`s using some custom functions. We should use a proper pretty-printing library and be agnostic about the final string type. Precedence and associativity of operators should also be taken into account.

I've started writing a suitable library (mostly by extracting various pieces from the `plutus-core` codebase).

**Comment by jakzale** [comment]
Add a separate pretty printing pass for golden testing.

**Comment by jakzale** [comment]
@effectfully was the pretty printing library released?

**Comment by effectfully** [comment]
Nope, but we can directly depend on the [`prettyprinter-configurable`](https://github.com/input-output-hk/plutus/tree/master/prettyprinter-configurable) subfolder of the Plutus repo.

**Comment by jakzale** [comment]
Urgh, got hit by https://github.com/commercialhaskell/stack/issues/5004

Do you have anything against git submodules?
